pkg/util/refreshable/refreshable.go

```diff
@@ -16,6 +16,7 @@ import (
 type Authorizer interface {
 	autorest.Authorizer
 	RefreshWithContext(context.Context, *logrus.Entry) (bool, error)
+	RefreshWithContextRawError(context.Context, *logrus.Entry) error
```

Comment:
Ah, right. When I [proposed this originally](https://github.com/Azure/ARO-RP/pull/2202#discussion_r927624201) it looks like I forgot about the fact that `RefreshWithContext` does the error skipping.

Any ideas on how we can make it work without an additional almost idential method?

Reply:
The problem here is that there are 2 different `RefreshWithContext` from what I can see . The one from the interface does more complicated things then the one from the `sp *adal.ServicePrincipalToken` (it calls it but does extra stuff with it). 

Possible solution would probably be to separate the call to `a.sp.RefreshWithContext` and add a function which does `error -> bool, error`. This way we don't need the `RefreshWithContextRawError`.

Reply:
@facchettos : the problem here is RefreshWithContext should return nil on certain errors and we are trying to capture one among them to send proper error message to customer, otherwise it will end up in the generic error (Internal Server Error) which is hard to understand the issue.
also didnt get your solution with respect to the current scenario